### ISSUE TITLE
UTF-8 special character produce an error in templates

### ISSUE THREAD

**Issue by techws-dev** [getgrav/grav#3716]
I upgraded to new version of Grav, and I have this error in some of my pages:

```
iconv(): Detected an illegal character in input string
```

I use utf-8 smileys in my pages, and a recent change in /system/src/Grav/Common/Helpers/Truncator.php is now throwing this error.

**Comment by rhukster** [comment]
Yes, i'm aware and currently looing for a solution.  I tried the recommended replacement for `mb_convert_encoding()` with HTML_ENTITIES, as it's deprecated in PHP 8.2 and being removed in PHP 9.  Unfortunately it doesn't work with extended characters such as smiley's and some other UTF-8 chars such as ⌘.

I will get a patched released out ASAP.

**Comment by rhukster** [comment]
please try replacing line `147` of the Truncator.php file with:

```
$html = mb_encode_numericentity($html, [0x80, 0x10FFFF, 0, ~0], 'UTF-8');
```
